locations/spiders/hannaford.py

```diff
@@ -1,122 +1,30 @@
-import re
+import json
 
-import scrapy
+from scrapy.spiders import SitemapSpider
 
-from locations.items import GeojsonPointItem
+from locations.hours import OpeningHours
+from locations.structured_data_spider import StructuredDataSpider
 
-regex = r"(\D+,\s\D+-\d{5}).*"  # city, state, zip
-regex_am = r"\s?([Aa][Mm])"
-regex_pm = r"\s?([Pp][Mm])"
-regex_lat = r"(lat = \d+\.\d+)"
-regex_lon = r"(lng = -\d+\.\d+)"
-regex_id = r"(t\.id = \"\d+\")"
 
-
-class HannafordSpider(scrapy.Spider):
+class HannafordSpider(SitemapSpider, StructuredDataSpider):
     name = "hannaford"
     item_attributes = {"brand": "Hannaford"}
```

Comment:
```suggestion
    item_attributes = {"brand": "Hannaford", "brand_wikidata": "Q5648760"}
```

I don't know if you're involved in the OSM or Wikidata world, if not don't worry. We try have Wikidata QIDs here so we can interoperate with other projects. That way we can get the category from [NSI](https://nsi.guide/index.html?t=brands&k=shop&v=supermarket&tt=Q5648760#hannaford-dde59d), or other metadata like social media links from [Wikidata](https://www.wikidata.org/wiki/Q5648760)